src/common/wui_api.c

```diff
@@ -286,10 +286,53 @@ void sntp_set_system_time(uint32_t sec) {
     sntp_time_init = true;
 }
 
-uint32_t stringify_timestamp(time_str_t *dest, timestamp_t *timestamp) {
+uint32_t stringify_timestamp(time_str_t *dest, timestamp_t *timestamp, uint8_t flag) {
     if (sntp_time_init) {
-        snprintf(dest->time, MAX_TIME_STR_SIZE, "%02d:%02d:%02d", timestamp->time.h, timestamp->time.m, timestamp->time.s);
-        snprintf(dest->date, MAX_DATE_STR_SIZE, "%02d.%02d.%d", timestamp->date.d, timestamp->date.m, timestamp->date.y);
+
+        dest->time[0] = 0;
+        dest->date[0] = 0;
+
+        // TIME PARSING
+        if (flag & TIME_STR_HOURS) {
+            snprintf(dest->time, MAX_TIME_STR_SIZE, "%02d", timestamp->time.h);
+        }
+        if (flag & TIME_STR_MINS) {
+            uint8_t length = strlen(dest->time);
+            if (length > 0) {
+                snprintf(dest->time + length, MAX_TIME_STR_SIZE - length, ":%02d", timestamp->time.m);
+            } else {
+                snprintf(dest->time, MAX_TIME_STR_SIZE, "%02d", timestamp->time.m);
+            }
+        }
+        if (flag & TIME_STR_SECS) {
+            uint8_t length = strlen(dest->time);
+            if (length > 0) {
+                snprintf(dest->time + length, MAX_TIME_STR_SIZE - length, ":%02d", timestamp->time.s);
+            } else {
+                snprintf(dest->time, MAX_TIME_STR_SIZE, "%02d", timestamp->time.s);
+            }
+        }
+
+        // DATE PARSING
+        if (flag & TIME_STR_DAYS) {
```

Comment:
do you realize this is not a valid English date format?

Reply:
i would use standard time library
http://www.cplusplus.com/reference/ctime/strftime/
it supports national mutations
http://zetcode.com/articles/cdatetime/

Reply:
commit `187f789` ISO format, (We can reimplement to strftime anytime in the future, for now I use my own timestamp structure and my own print function)